gossipsub.go

```diff
@@ -180,8 +210,14 @@ func (gs *GossipSubRouter) HandleRPC(rpc *RPC) {
 }
 
 func (gs *GossipSubRouter) handleIHave(p peer.ID, ctl *pb.ControlMessage) []*pb.ControlIWant {
-	iwant := make(map[string]struct{})
+	// we ignore IHAVE gossip from any peer whose score is below the gossip threshold
+	score := gs.score.Score(p)
+	if score < gs.gossipThreshold {
```

Comment:
The gossip threshold is generally negative, so that we allow for some gossip disemination from low scoring peers. If the score drops below that, the peer is effectively graylisted until his negative score decays (if it does that).
So what you are asking for is exactly what is happening.